### PR TITLE
Test of post collection handler

### PR DESCRIPTION
**Pull request by raonismaneoto**
#98

### CODE REVIEW

File: test/post_collection_handler_test.py [hunk 1]
```diff
@@ -0,0 +1,150 @@
+# -*- coding: utf-8 -*-
+"""Post handler test."""
```
Comment (mayzabeel) [inline test/post_collection_handler_test.py]: Post Collection handler test

File: test/post_collection_handler_test.py [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+
+from test_base import TestBase
+from models.user import User
+from models.institution import Institution
+from handlers.post_collection_handler import PostCollectionHandler
+from google.appengine.ext import ndb
+import json
+
+
+class PostHandlerTest(TestBase):
```
Comment (mayzabeel) [inline test/post_collection_handler_test.py]: PostCollectionHandlerTest

File: test/post_collection_handler_test.py [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+    """Test the post_handler class."""
```
Comment (mayzabeel) [inline test/post_collection_handler_test.py]: Post Collection handler test

File: test/post_collection_handler_test.py [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+        self.mayza = self.mayza.key.get()
+        # Check if the post's key is in institution and user
+        self.assertTrue(key_post in self.mayza.posts,
+                        "The post is not in user.posts")
+        self.assertTrue(key_post in self.certbio.posts,
+                        "The post is not in institution.posts")
+        # Check if the post's attributes are the expected
+        self.assertEqual(post_obj.title, 'new post',
+                         "The title expected was new post")
+        self.assertFalse(post_obj.title == 'newpost',
```
Comment (mayzabeel) [inline test/post_collection_handler_test.py]: Basta o assert acima para o title. Pode remover esse

File: test/post_collection_handler_test.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+        self.raoni = self.raoni.key.get()
+        # Check if the post's key is in institution and user
+        self.assertTrue(key_post in self.raoni.posts,
+                        "The post is not in user.posts")
+        self.assertTrue(key_post in self.certbio.posts,
+                        "The post is not in institution.posts")
+        # Check if the post's attributes are the expected
+        self.assertEqual(post_obj.title, 'another post',
+                         "The title expected was another post")
+        self.assertFalse(post_obj.title == 'anotherpost',
```
Comment (mayzabeel) [inline test/post_collection_handler_test.py]: remove esse também.

File: test/post_collection_handler_test.py [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+                         "The title expected was new post")
+        self.assertFalse(post_obj.title == 'newpost',
+                         "The title shouldn't be newpost")
+        self.assertEqual(post_obj.institution, self.certbio.key,
+                         "The post's institution is not the expected one")
+        self.assertEqual(post_obj.text,
+                         'testing new post',
+                         "The post's text is not the expected one")
+
+        self.os.environ['REMOTE_USER'] = 'raoni.smaneoto@ccc.ufcg.edu.br'
```
Comment (mayzabeel) [inline test/post_collection_handler_test.py]: Basta testar uma vez, se esse another post não tem nenhum comportamento diferente do primeiro não precisa repetir o teste :)

File: test/post_collection_handler_test.py [hunk 7]
```diff
@@ -0,0 +1,150 @@
...
+class PostHandlerTest(TestBase):
+    """Test the post_handler class."""
+
+    @classmethod
+    def setUp(cls):
+        """Provide the base for the tests."""
+        cls.test = cls.testbed.Testbed()
+        cls.test.activate()
+        cls.policy = cls.datastore.PseudoRandomHRConsistencyPolicy(
+            probability=1)
```
Comment (andreldsa) [inline test/post_collection_handler_test.py]: pq probability igual a 1 e não 0?

> The PseudoRandomHRConsistencyPolicy class lets you control the likelihood of a write applying before each global (non-ancestor) query. By setting the probability to 0%, we are instructing the datastore stub to operate with the maximum amount of eventual consistency. Maximum eventual consistency means writes will commit but always fail to apply, so global (non-ancestor) queries will consistently fail to see changes. This is of course not representative of the amount of eventual consistency your application will see when running in production, but for testing purposes, it's very useful to be able to configure the local datastore to behave this way every time. If you use a non-zero probability, PseudoRandomHRConsistencyPolicy makes a deterministic sequence of consistency decisions so test outcomes are consistent

[https://cloud.google.com/appengine/docs/standard/python/tools/localunittesting](url)

**Pull request merged**